articles/stripe-integration-react/index.md

```diff
@@ -0,0 +1,139 @@
+1. Card details are entered on the website.
+2. The web application sends a request to the Stripe API with card details in it.
+3. Stripe API stores the card details securely and sends a token to the web application.
+4. The token is a point of reference to identify the card. Note that the card details are never shared with anyone for security reasons. The backend server receives the token and other meta-data related to the purchase.
+5. The backend application makes a request to the API with the token in place. Also, it's a good practice to recalculate the price at the backend, as the token does not contain information about the total amount to be charged.  
+
+![Payment Flow](/engineering-education/stripe-integration-react/paymentflow.jpg)
+[*Image Source:*](https://stripe.com/docs/payments/accept-a-payment#web)
+
+With this information in mind, we can understand the important role Stripe plays. Payment gateways such as Stripe and PayPal garner the trust of millions of users by providing seamlessly integrated solutions.
+
+#### Creating a Stripe Account
+To create an account, [register for an account](https://dashboard.stripe.com/register?redirect=%2Ftest%2Fpayments) and provide your account details (email, mobile number, name, and password). Upon successful registration, a confirmation email is sent to the registered email account. Click on the email link to activate your new Stripe account.
+
+If already registered, log in to your dashboard with your credentials.
+
+#### Switching to Test Mode on Stripe
+This step is important to ensure we are in testing mode. This enables us to test aspects of payments such as receiving payments, sending payments, or approving refunds. On the bottom left of your dashboard, you should find the **Viewing test data** button. Enable that by toggling the button provided. The dashboard is shown for your reference.
+
+![Dashboard View](/engineering-education/stripe-integration-react/dashboard.jpg)
+*Dashboard View*
+
+#### Obtain a Stripe Access Token
+You will find a section called `Get your test API keys`. Under there, you will find two keys, a **Publishable key**, and a **Secret key**. As the name suggests, we need to keep these keys secure. The publishable key is used to send requests, whereas the secret key is stored in the `.env` file at the backend.
+
+![API Test Keys](/engineering-education/stripe-integration-react/api_test_keys.jpg)
+*API Keys on Dashboard*
+
+### Stripe Integration using react-stripe-checkout
+There are many libraries available for integrating React with stripe. We will be using the `react-stripe-checkout` library in this tutorial.
+
+Install the library using the following command:
+
+```
+npm install react-stripe-checkout
+```
+
+Once installed, create a new component called `CheckoutWithStripe.js` and add it to your main component list. Insert the appropriate details and create your custom functions to add more details to the API call.
```

Comment:
What are the appropriate details? Is there a list somewhere?

Reply:
While creating a react app, there are other components in the components folder. I was referring to that list of components.